tasks/main.yml

```diff
@@ -83,6 +94,25 @@
   when: docker_upstream|d() | bool
   tags: [ 'role::docker:config' ]
 
+- name: Configure Docker proxy
+  template:
+    src: 'etc/systemd/system/docker.service.d/http-proxy.conf.j2'
+    dest: '/etc/systemd/system/docker.service.d/http-proxy.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: [ 'Restart docker' ]
+  register: docker_register_systemd_service
```

Comment:
I would use a separate variable name, for example `docker_register_systemd_proxy_present` to indicate changes to the file, and check all relevant variables in the `systemctl daemon-reload` task. Using the same variable names might result in wrong configuration in certain situations.